app/domain/authentication/authn_k8s/validate_pod_request.rb

```diff
@@ -72,7 +72,11 @@ def host_can_access_service?
       end
 
       def host
-        @host ||= @resource_class[k8s_host.conjur_host_id]
+        unless defined? @host
+          @host = @resource_class[k8s_host.conjur_host_id]
+        end
+        return @host unless @host.nil?
+        raise Err::HostNotFound(k8s_host.conjur_host_id)
```

Comment:
Btw, even though we haven't followed this in our ruby codebase everywhere, I do like the Go convention of "happy path to the left",  and there no reason not to use it in ruby as well, so here you could swap the order of these two lines.